cmd/save_test.go

```diff
@@ -131,12 +131,12 @@ func TestSaveRun(t *testing.T) {
 		{"no data", "me/bad_dataset", "", "", "", "", false, false, true, "", "no changes to save", ""},
 		{"bad dataset file", "me/cities", "bad/filpath.json", "", "", "", false, false, true, "", "open bad/filpath.json: no such file or directory", ""},
 		{"bad body file", "me/cities", "", "bad/bodypath.csv", "", "", false, false, true, "", "opening dataset.bodyPath 'bad/bodypath.csv': path not found", ""},
-		{"good inputs, dryrun", "me/movies", "testdata/movies/dataset.json", "testdata/movies/body_ten.csv", "", "", false, true, true, "dataset saved: peer/movies@/map/QmVDSrf4BpupXYmDJc6eXanFVW9DS6g3Pfkkx6YqkcUjEQ\nthis dataset has 1 validation errors\n", "", ""},
-		{"good inputs", "me/movies", "testdata/movies/dataset.json", "testdata/movies/body_ten.csv", "", "", true, false, true, "dataset saved: peer/movies@/map/QmWR7jojXb8pGuDPAtkk3FsTg2mkqkcbYWgVr6C8SzVupT\nthis dataset has 1 validation errors\n", "", ""},
-		{"add rows, dry run", "me/movies", "testdata/movies/dataset.json", "testdata/movies/body_twenty.csv", "Added 10 more rows", "Adding to the number of rows in dataset", false, true, true, "dataset saved: peer/movies@/map/QmPg2oEhBJSrJAVHgH7ptsmHQEVEio6JdcgHrDY45eYVtg\nthis dataset has 1 validation errors\n", "", ""},
-		{"add rows, save", "me/movies", "testdata/movies/dataset.json", "testdata/movies/body_twenty.csv", "Added 10 more rows", "Adding to the number of rows in dataset", true, false, true, "dataset saved: peer/movies@/map/QmTFfwHuWcXLbpL4b9Pb7BMM7zxhjqrgQRMiCSNub4ZX3E\nthis dataset has 1 validation errors\n", "", ""},
+		{"good inputs, dryrun", "me/movies", "testdata/movies/dataset.json", "testdata/movies/body_ten.csv", "", "", false, true, true, "dataset saved: peer/movies\n", "", ""},
+		{"good inputs", "me/movies", "testdata/movies/dataset.json", "testdata/movies/body_ten.csv", "", "", true, false, true, "dataset saved: peer/movies@/map/QmVDSrf4BpupXYmDJc6eXanFVW9DS6g3Pfkkx6YqkcUjEQ\nthis dataset has 1 validation errors\n", "", ""},
```

Comment:
what's breaking hashes on these tests? I'm assuming some sort of broken/bad/missing thing, just curious

Reply:
More subtle than that. The `dsfs.Timestamp` stub used by testRunner increments the minute by 1 every time it's called. The early return being added for dryRun (https://github.com/qri-io/qri/pull/1369/files#diff-8f260ff9153f41a66f62970bbc74ce9cR644) prevents `dsfs.Timestamp` from being called, which changes the timestamp and therefore hashes for future steps. If I were to add a call to `dsfs.Timestamp` just before that early return, the hash would stay the same. So the hashes changing is proof that no dsfs actions are occuring for dry-run.